content/articles/sentiment-analysis-with-spacy-and-scikit-learn/index.md

```diff
@@ -0,0 +1,472 @@
+Sentiment analysis is a subset of natural language processing and text analysis that detects positive or negative sentiments in a text. Sentiment analysis helps businesses to understand how people gauge and their feelings towards their brand and products.
+
+This is done using the reviews left on their sites and monitoring online conversations.
+
+Sentiment analysis is used to analyze customer feedback, this helps businesses to know if customers are happy or frustrated with their products. Businesses use this information to make changes in their products and provide the best products that meet customer's needs.
+
+In this tutorial, we shall use Spacy in building our sentiment analysis model, we shall use three datasets from IMDb, Amazon, and Yelp.
+These datasets contain various reviews that are either labeled positive or negative. This dataset contains movies reviews in the IMDb dataset, products bought reviews from the Amazon dataset, and local business and social networking site reviews from the Yelp dataset.
+
+These datasets will be able to build our model that can be able to classify a review as either positive or negative. This will help a business to know if customers are satisfied or disgruntled with their products or services. This helps the business to make the necessary adjustments to improve the products or services offered.
+
+### Table of contents
+
+- [Prerequisites](#prerequisites)
+- [Introduction](#introduction)
+- [Dataset used](#dataset-used)
+- [Loading dataset](#loading-dataset)
+- [Adding headers](#adding-headers)
+- [Merge or concatinate our datasets with the keys](#merge-or-concatinate-our-datasets-with-the-keys)
+- [Removing stopwords](#removing-stopwords)
+- [Loading machine learning packages](#loading-machine-learning-packages)
+- [Custom transformer class](#custom-transformer-class)
+- [Vectorization and classifier](#vectorization-and-classifier)
+- [TfidfVectorizer method](#tfidfVectorizer-method)
+- [Features and labels](#features-and-labels)
+- [Dataset splitting](#dataset-splitting)
+- [Creating the pipeline](#creating-the-pipeline)
+- [Accuracy score](#accuracy-score)
+- [Making prediction](#making-prediction)
+- [Conclusion](#Conclusion)
+- [References](#References)
+
+### Prerequisites
+
+1. A good understanding of [Python](/engineering-education/python-projects-for-beginners/).
+2. A good working knowledge of [machine learning](/engineering-education/house-price-prediction/).
+3. Have some working knowledge of [natural language processing.](https://en.wikipedia.org/wiki/Natural_language_processing)
+
+> NOTE: To follow along easily, use [Google Colab](https://research.google.com/).
+
+### Introduction
+
+There are different types of sentiment analysis, we have those that focus on the polarity of a given text positive, neutral, and negative, others focus on feeling and emotions within a text, and other focus on the intentions and urgency of a customer. They are grouped as follows.
```

Comment:
This is a complex sentence. Consider breaking it down into 2 or more simpler sentences.